### PR TITLE
Fix no default columns

### PR DESCRIPTION
**Pull request by geloescht**
## Description of changes

Move around order of operations in admin/client/App/screens/List/index.js and admin/client/utils/List.js to fix display of lists without default or name column.

## Related issues (if any)

#2945 

## Testing

- [x] Please confirm `npm run test-all` ran successfully.

### CODE REVIEW

File: test/e2e/models/misc/NoDefaultColumn.js [hunk 1]
```diff
@@ -4,11 +4,6 @@ var Types = keystone.Field.Types;
 // Model to demonstrate issue #2945
```
Comment (jstockwin) [inline test/e2e/models/misc/NoDefaultColumn.js]: @geloescht This comment should probably be removed. Otherwise LGTM.

**Comment by mxstbr** [comment]
LGTM too

**Comment by jstockwin** [comment]
New descriptions/comments LGTM. Nice one putting it into the lists group too, that makes much more sense. @mxstbr? :shipit:

**Comment by mxstbr** [comment]
LGTM :shipit:

**Comment by JedWatson** [comment]
Nice one @geloescht!

**Pull request merged**